### PR TITLE
added a "Split pile" feature to the pile "context menu"

### PR DESCRIPTION
**Pull request by ArnoldSmith86**
It should behave similar to the one that was added to PCIO a few weeks ago.

### CODE REVIEW

**Comment by ArnoldSmith86** [comment]
PR-SERVER-BOT: You can play around with it here: http://212.47.248.129:3049/pr-test (or any other room on that server)

**Comment by RaphaelAlvez** [comment]
would it be difficult for you to add a /(total amount) fixed on the left?

As we have labels on the bottom the problem with splitting at the top of the boards is even more significant. I suggest adding a condition that split to the bottom only if the new pile would be totally above the board

**Comment by RaphaelAlvez** [comment]
tested you changes

image

As you can see in the image there is the possibility of the new label to be under the old pile. Either put the new pile always on top or use the two left images as a reference and only apply the bottom slipt when the new label would go to the bottom.

**Comment by ArnoldSmith86** [comment]
I've been playing with a few variations and I don't think there's an easy fix for that.

**Comment by RaphaelAlvez** [comment]
> I've been playing with a few variations and I don't think there's an easy fix for that.

Agree after more testing it seems difficult to determine a good solution.

Have you tried always putting the new pile on top? That should at least be better than the current situation.
I assume that if I slipt 4 cards, I would want to use those 4 cards

**Comment by ArnoldSmith86** [comment]
I was emulating PCIO behavior. We could change it of course. What do others think about it?

File: client/js/widgets/pile.js [hunk 1]
```diff
@@ -74,6 +74,33 @@ class Pile extends Widget {
...
+    const childCount = this.children().length;
+    const countDiv = document.createElement('div');
+    countDiv.textContent = `/ ${childCount}`;
+    $('#pileOverlay').appendChild(countDiv);
+    const splitInput = document.createElement('input');
+    splitInput.type = 'number';
+    splitInput.value = Math.floor(childCount/2);
+    splitInput.min = 0;
+    splitInput.max = childCount;
+    countDiv.prepend(splitInput);
```
Comment (UltimateGeek) [inline client/js/widgets/pile.js:86]: could a label "Split: " be included before the input control?

**Review by UltimateGeek — COMMENTED** [review]
* [ ] update wiki once PR merged

**Comment by RaphaelAlvez** [comment]
@ArnoldSmith86 not giving my approval until the pile with the number of cards i typed in is on top.
I approve the feature though.

**Review by UltimateGeek — APPROVED** [review]
I tested the card order when splitting using 4 cards. They stayed in the same order.

**Comment by ArnoldSmith86** [comment]
> @ArnoldSmith86 not giving my approval until the pile with the number of cards i typed in is on top.
> I approve the feature though.

Is this good enough now?

**Review by robartsd — COMMENTED** [review]
I think split should take the top x cards from the pile and move them leaving them at a higher z-index than the cards remaining in the starting position. Current test moves the remaining cards in the pile, leaving the selected number of cards in the original position (but on top as Arnold requested). Moving the bottom of the pile is convenient for cutting the deck (just need to move the cards from the bottom on top of the other cards). Perhaps there should be a "from the bottom" option checkbox to split off the bottom x cards instead of the top x cards.

**Comment by ArnoldSmith86** [comment]
I still want to change this based on the feedback BTW.

**Review by RaphaelAlvez — APPROVED** [review]
Tested and it's perfect now

**Review by robartsd — APPROVED** [review]
Works as expected.

**Pull request merged**